pkg/ccl/multitenantccl/tenantcostclient/tenant_side_test.go

```diff
@@ -968,6 +971,98 @@ func TestSQLLivenessExemption(t *testing.T) {
 	)
 }
 
+// TestScheduledJobsConsumption verifies that the scheduled jobs system itself
+// does not consume RUs, but that the jobs it runs do consume RUs.
+func TestScheduledJobsConsumption(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	ctx := context.Background()
+	hostServer, _, _ := serverutils.StartServer(t, base.TestServerArgs{})
+	defer hostServer.Stopper().Stop(ctx)
+
+	st := cluster.MakeTestingClusterSettings()
+	stats.AutomaticStatisticsClusterMode.Override(ctx, &st.SV, false)
+	tenantcostclient.TargetPeriodSetting.Override(ctx, &st.SV, time.Millisecond*20)
+
+	testProvider := newTestProvider()
+
+	// Delay jobs from being scheduled until we're ready.
+	delayCh := make(chan struct{}, 0)
+	scanDelay := func() time.Duration {
+		<-delayCh
+		return 0
+	}
+
+	env := jobstest.NewJobSchedulerTestEnv(jobstest.UseSystemTables, timeutil.Now())
+	var zeroDuration time.Duration
+	tenantServer, tenantDB := serverutils.StartTenant(t, hostServer, base.TestTenantArgs{
+		TenantID:                    serverutils.TestTenantID(),
+		Settings:                    st,
+		AllowSettingClusterSettings: true,
+		TestingKnobs: base.TestingKnobs{
+			TenantTestingKnobs: &sql.TenantTestingKnobs{
+				OverrideTokenBucketProvider: func(kvtenant.TokenBucketProvider) kvtenant.TokenBucketProvider {
+					return testProvider
+				},
+			},
+			TTL: &sql.TTLTestingKnobs{
+				// Don't wait until we can use a historical query.
+				AOSTDuration: &zeroDuration,
+			},
+			JobsTestingKnobs: &jobs.TestingKnobs{
+				JobSchedulerEnv:                 env,
+				SchedulerDaemonInitialScanDelay: scanDelay,
```

Comment:
You might find TakeOverJobScheduling more convenient -- just returns you a function you can invoke directly, without depending on any pacing logic in the scheduler.

Reply:
OK, I switched to that.